client/my-sites/themes/single-site.jsx

```diff
@@ -37,30 +37,31 @@ const ThemesSingleSite = ( props ) => {
 		{ analyticsPath, analyticsPageTitle, isJetpack, translate } = props,
 		jetpackEnabled = config.isEnabled( 'manage/themes-jetpack' );
 
-	if ( isJetpack && jetpackEnabled && ! site.hasJetpackThemes ) {
-		return <JetpackUpgradeMessage site={ site } />;
-	}
-
-	if ( isJetpack && jetpackEnabled && ! site.canManage() ) {
-		return <JetpackManageDisabledMessage site={ site } />;
-	}
-
-	if ( isJetpack && ! jetpackEnabled ) {
-		return (
-			<Main className="themes">
-				<PageViewTracker path={ analyticsPath } title={ analyticsPageTitle }/>
-				<SidebarNavigation />
-				<CurrentTheme
-					site={ site }
-					canCustomize={ site && site.isCustomizable() } />
-				<EmptyContent title={ translate( 'Changing Themes?' ) }
-					line={ translate( 'Use your site theme browser to manage themes.' ) }
-					action={ translate( 'Open Site Theme Browser' ) }
-					actionURL={ site.options.admin_url + 'themes.php' }
-					actionTarget="_blank"
-					illustration="/calypso/images/drake/drake-jetpack.svg" />
-			</Main>
-		);
+	if ( isJetpack ) {
```

Comment:
> since this is currently a stateless functional component

Oh right! I didn't see that before. Well, it's already one level of indentation better. I think my eyes start to hurt after 3. I'll say that's fine with me for now to keep it from having to be an object.

I haven't used functional components for more than three-line components before. This is an interesting exercise in how to compress complicated logic into small spaces. I suppose you could define additional render methods as separate functions within the closure, but that feels maybe a little awkward? Are there any patterns I could learn here?